hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -57,6 +57,14 @@
     except ImportError:
         _AnnotatedAlias = ()
 
+try:
+    from typing import TypeAlias  # type: ignore
+except ImportError:
+    try:
+        from typing_extensions import TypeAlias
+    except ImportError:
+        TypeAlias = object()  # It must be a singleton
```

Comment:
Tricky edge case here: we want to handle `typing_extensions.TypeAlias`, even when `typing.TypeAlias` exists, so I don't think we need this import logic.

(Since the tests passs it looks like this is just ~~an alias~~ re-exported from `typing` if available, but I'd rather not assume that in our implementation)

Reply:
Good catch 👍